docs/README.md

```diff
@@ -1,49 +1,60 @@
+
+### Requirements for adding a new load generators
+1. Create a docker image that must contain an executable of a new load generator and all required scripts to run it. Docker image should exit once load test is finished and it should provide logs to stdout which will be used by Kangal Proxy.
 
-### Adding a new load generator
-Kangal offers an opportunity to add different load generators as backends. 
-Requirements to new load generators:
-1. Create a docker image that must contain an executable of a new load generator and all required scripts to run it. Docker image should exit once load test is finished and it should provide logs to stdout which will be used by Kangal Proxy API.
 2. Create a new backend resource definition in Kangal source code: 
- - [/pkg/backends](https://github.com/hellofresh/kangal/tree/master/pkg/backends). 
- - [backend.go](https://github.com/hellofresh/kangal/blob/master/pkg/backends/backend.go#L33)
- - [CRD definition](https://github.com/hellofresh/kangal/blob/master/charts/kangal/crd.yaml#L43)
- - [openapi.json](https://github.com/hellofresh/kangal/blob/master/openapi.json#L280)
-The basic resource is a job that manages all the other resources and sets pods to the `finished` state when the test is over.
+ - [pkg/backends/](/pkg/backends)
+ - [pkg/backends/backend.go](/pkg/backends/backend.go#L33)
+ - [charts/kangal/crd.yaml](/charts/kangal/crd.yaml#L43)
+ - [openapi.json](/openapi.json#L280)
 
-## Reporting in Kangal
+## Reporting
 Reporting is an important part of load testing process. It basically contains in two parts:
-1. Live metrics during the running load test - Kangal proxy scrapes logs from main job stdout Docker container.
+
+1. Live metrics during the running load test, Kangal Proxy scrapes logs from main job stdout container.
 2. Solid report generated after the end of the test. 
-Currently, Kangal relies on report creation implemented in backend. You can read more about JMeter backend implementation in [Reporting in JMeter](jmeter-in-kangal/Reporting-in-JMeter.md).
 
-To persist the reports the backend receives a Pre-Signed URL where which can use to upload the report. If the report contains multiple files it will be necessary to archieve/compress into a single file.
+Kangal Proxy provides an API endpoint that allows to retrieve persisted reports (`/load-test/:name/report/`).
+
+> Kangal relies on report creation to be implemented in the backend.
 
-Kangal also provides an API endpoint that allows to retrieve reports from storage bucket.
+### Persisting reports
+Kangal generates a Pre-Signed URL and backend can use it to persist a report.
+
+> If the report contains multiple files it will be necessary to archieve/compress into a single file.
 
 To allow Kangal to serve the report static files is necessary to explicit set the file as a `tar` archive with no compression and **no enclosing directory**, otherwise the endpoint will just force the report download.
```

Comment:
```suggestion
To allow Kangal to serve the report static files it is necessary to explicitly set the file as a `tar` archive with no compression and **no enclosing directory**, otherwise, the endpoint will just force the report download.
```